maven/bnd-export-maven-plugin/src/main/java/aQute/bnd/maven/export/plugin/ExportMojo.java

```diff
@@ -87,78 +61,10 @@ private void export(File runFile) throws MojoExecutionException, Exception, IOEx
 				throw new MojoExecutionException("Initializing the workspace failed " + run.getErrors());
 			}
 			if (resolve) {
-				resolve(run);
-			}
-			export(run);
-		}
-	}
-
-	private void export(Run run) throws Exception {
-		try (Jar jar = run.getProjectLauncher().executable()) {
-			targetDir.mkdirs();
-			File jarFile = new File(targetDir, getNamePart(run.getPropertiesFile()) + ".jar");
-			jar.write(jarFile);
-		}
-	}
-
-	private void resolve(StandaloneRun run) throws Exception, IOException {
-		try (ProjectResolver projectResolver = new ProjectResolver(run)) {
-			Map<Resource,List<Wire>> resolution = projectResolver.resolve();
-			Set<Resource> resources = resolution.keySet();
-			List<VersionedClause> runBundles = new ArrayList<>();
-			for (Resource resource : resources) {
-				VersionedClause runBundle = ResourceUtils.toVersionClause(resource, "[===,==+)");
-				if (!runBundles.contains(runBundle)) {
-					runBundles.add(runBundle);
-				}
-			}
-			Collections.sort(runBundles, new Comparator<VersionedClause>() {
-				@Override
-				public int compare(VersionedClause a, VersionedClause b) {
-					int diff = a.getName().compareTo(b.getName());
-					return (diff != 0) ? diff : a.getVersionRange().compareTo(b.getVersionRange());
-				}
-			});
-
-			File runFile = run.getPropertiesFile();
-			BndEditModel bem = new BndEditModel(run.getWorkspace());
-			Document doc = new Document(IO.collect(runFile));
-			bem.loadFrom(doc);
-
-			List<VersionedClause> bemRunBundles = bem.getRunBundles() != null ? bem.getRunBundles()
-					: new ArrayList<VersionedClause>();
-
-			String originalRunbundlesString = headerClauseListFormatter.convert(bemRunBundles);
-			logger.debug("Original -runbundles was:\n\t {}", originalRunbundlesString);
-			String runbundlesString = headerClauseListFormatter.convert(runBundles);
-			logger.debug("Resolved -runbundles is:\n\t {}", runbundlesString);
-
-			List<VersionedClause> deltaAdd = new ArrayList<>(runBundles);
-			deltaAdd.removeAll(bemRunBundles);
-			List<VersionedClause> deltaRemove = new ArrayList<>(bemRunBundles);
-			deltaRemove.removeAll(runBundles);
-			boolean added = bemRunBundles.addAll(deltaAdd);
-			boolean removed = bemRunBundles.removeAll(deltaRemove);
-			if (added || removed) {
-				if (failOnChanges && !bemRunBundles.isEmpty()) {
-					String message = String.format(
-							"The runbundles have changed. Failing the build!\nWas: %s\nIs: %s",
-							originalRunbundlesString, runbundlesString);
-					throw new MojoExecutionException(message);
-				}
-				bem.setRunBundles(bemRunBundles);
-				logger.info("Writing changes to {}", runFile.getAbsolutePath());
-				logger.info("{}:{}", Constants.RUNBUNDLES, bem.getDocumentChanges().get(Constants.RUNBUNDLES));
-				bem.saveChangesTo(doc);
-				IO.store(doc.get(), runFile);
+				run.resolve(failOnChanges, false);
 			}
+			run.export(targetDir);
```

Comment:
Need to make sure resolve output, if any, from above is used. 

Also need to check run object for errors/warnings after export call and report.